contracts/Staking.sol

```diff
@@ -820,17 +824,34 @@ contract Staking is Pauser, Whitelist {
         epochIdx = _epochIdx;
         currentEpochIdx = _epochSigners;
         currentVotingPowers = _epochVotingPowers;
+    }
+
+    function getCurrentEpoch()
+        public
+        view
+        returns (
+            uint256,
+            address[] memory,
+            uint256[] memory
+        )
+    {
+        return (epochIdx, currentEpochIdx, currentVotingPowers);
+    }
 
+    function proposalValidators() public view returns (address[] memory, uint256[] memory) {
         uint256 _maxBondedValidators = params[dt.ParamName.MaxBondedValidators];
         address[] memory _proposedValidators = new address[](_maxBondedValidators);
         uint256[] memory _proposedVotingPowers = new uint256[](_maxBondedValidators);
         for (uint256 i = 0; i < _maxBondedValidators; i++) {
             if (validators[bondedValAddrs[i]].tokens != 0) {
                 _proposedValidators[i] = validators[bondedValAddrs[i]].signer;
-                _proposedVotingPowers[i] = validators[bondedValAddrs[i]].tokens;
+                // we assume minimal stake is also 1e18
+                require(validators[bondedValAddrs[i]].tokens > STAKE_UINT, "minimal stake is 1e18");
```

Comment:
The method should never fail.  That said, we should either skip the validator and comment something else that this should never happen?